tests/frontend/org/voltdb/newplanner/TestCalciteLogicalRules.java

```diff
@@ -0,0 +1,418 @@
+/* This file is part of VoltDB.
+ * Copyright (C) 2008-2018 VoltDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.voltdb.newplanner;
+
+import org.apache.calcite.plan.RelOptUtil;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.RelRoot;
+import org.voltdb.catalog.org.voltdb.calciteadaptor.CatalogAdapter;
+import org.voltdb.newplanner.rules.PlannerPhase;
+import org.voltdb.types.CalcitePlannerType;
+
+public class TestCalciteLogicalRules extends PlanRulesTestCase {
+    @Override
+    protected void setUp() throws Exception {
+        setupSchema(TestVoltSqlValidator.class.getResource(
+                "testcalcite-ddl.sql"), "testcalcite", false);
+        init(CatalogAdapter.schemaPlusFromDatabase(getDatabase()));
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Override
+    void assertPlanMatch(String sql, String expectedPlan) {
+        RelRoot root = parseValidateAndConvert(sql);
+        RelNode node = CalcitePlanner.transform(CalcitePlannerType.HEP, PlannerPhase.CALCITE_LOGICAL,
+                root.rel);
+        String actualPlan = RelOptUtil.toString(node);
+        assertEquals(expectedPlan, actualPlan);
+    }
+
+    public void testSimpleSeqScan() {
```

Comment:
A better way to organize these tests would probably be like:
```suggestion
    final Map<String, Pair<String, String>> testQueries = // Test topic => (query, expected plan)
         new HashMap<String, Pair<String, String>>() {{
            put("simple sequential scan", Pair.of("SELECT si FROM Ri1", "LogicalCalc..."));
            ...
            }};
            final failedTests = testQueries.stream().filter(kv -> {
               final String topic = kv.getKey();
               final Pair<String, String> test = kv.getValue();
               return ! planOf(test.getFirst()).equals(test.getSecond());
               }).collect(Collectors.toMap());
               if (! failedTests.isEmpty()) {
                  // report each, and call fail()
                }
            
```

Reply:
No. I want to split the test cases. 
Usually, I need to test&debug on a single test case.
Although it looks ugly but easy to maintenance.